### DISCUSSION TITLE
penetration and normal

### DISCUSSION THREAD

**Discussion by 313364973** [dimforge/rapier#645] [Q&A]
hi,i only want to collide by myself，like one box collide to other，i want to know the collide point and collide penetration  and normal,so

**Comment by 313364973** [comment]
i can  write code to split the two

**Comment by ThierryBerger** [comment]
An option you have is to react to collision events, more info here: https://rapier.rs/docs/user_guides/rust/advanced_collision_detection#collision-and-contact-force-events
